test/typedef_test.dart

```diff
@@ -196,4 +197,22 @@ typedef R3<T> = R2<List<T>>;
       '</p>',
     );
   }
+
+  void test_typedefRetainsAliasWhenUsed() async {
+    var library = await bootPackageWithLibrary('''
+typedef R<T> = (T, String);
+
+R<int> f(int a, String b) {
+  return R<int>(a, b);
+}
+''');
+    final rTypedef = library.typedefs.named('R');
+    final fFunc = library.functions.named('f');
+
+    expect(fFunc.modelType.returnType, isA<AliasedUndefinedElementType>());
+    expect(
+        (fFunc.modelType.returnType as AliasedUndefinedElementType)
```

Comment:
This can be folded into the above expectation with [`having`](https://pub.dev/documentation/matcher/latest/matcher/TypeMatcher/having.html), I think.

```dart
isA<...>()
    .having((e) => e.typeAliasElement, 'typeAliasElement', equals(rTypedef.element))

Reply:
Ah, nice.  Much better.  Done.